lib/touch-enroll.js

```diff
@@ -65,29 +65,41 @@ async function getUserDefault (domain, key) {
 
 /**
  * Sets a MacOS User Default value on a domain
- * @param {*} domain 
- * @param {*} key 
- * @param {*} value 
+ * @param {*} domain
+ * @param {*} key
+ * @param {*} value
  */
 async function setUserDefault (domain, key, value) {
   await exec('defaults', ['write', 'Apple Global Domain', domain, '-dict-add', key, typeof(value) === 'undefined' ? 'nil' : value]);
 }
 
+/**
+ * Backup current values of user defaults and store them into an internal variable.
+ * Only one backup can be stored. All the further calls to this method will be ignored unless
+ * {#restoreTouchEnrollShortcuts} is called.
+ */
 async function backupTouchEnrollShortcuts () {
-  if (!touchEnrollBackups) {
-    touchEnrollBackups = await getTouchEnrollKeys();
+  if (!touchEnrollBackupsStack.length) {
+    touchEnrollBackupsStack.push(await getTouchEnrollKeys());
   }
 }
 
+/**
+ * Restore the user defaults previously preserved by {#backupTouchEnrollShortcuts}.
+ * Only one backup can be preserved. Multiple calls to this method
+ * will have no effect if the backup has already been restored or hasn't been made.
+ */
 async function restoreTouchEnrollShortcuts () {
-  if (touchEnrollBackups) {
-    await setTouchEnrollKeys(touchEnrollBackups);
-    touchEnrollBackups = undefined;
+  if (touchEnrollBackupsStack.length) {
+    const preservedKeys = touchEnrollBackupsStack.pop();
+    if (!touchEnrollBackupsStack.length) {
```

Comment:
I don't understand this. Only a single element will ever be put in the stack, so once we pop either `preservedKeys` has something in it, or is `undefined` (if you `pop` from the empty stack). Shouldn't that be checked, instead of the `length`, which can only be `0`?